libcaf_core/caf/span.hpp

```diff
@@ -174,6 +174,12 @@ class span {
   size_t size_;
 };
 
+template <class T>
+bool operator<(const span<T>& lhs, const span<T>& rhs) {
+  return std::lexicographical_compare(lhs.begin(), lhs.end(), rhs.begin(),
+                                      rhs.end());
+}
+
```

Comment:
If you need this in a test, please add a function for this in the test and remove this again. `std::span` doesn't support this, so neither do we.